core/src/main/scala/org/ensime/indexer/FileWatchers.scala

```diff
@@ -79,10 +144,10 @@ class ClassfileWatcher(
         running = scheduler.schedule(resetter, 1, TimeUnit.SECONDS)
       }
 
-      private val targets: Set[FileName] = for {
-        dir <- config.targetClasspath
+      private val targets: Set[FileName] = (for {
+        dir <- watchedDirs
         ref = vfs.vfile(dir)
-      } yield ref.getName()
+      } yield ref.getName()).toSet
```

Comment:
no need! It is actually a `Set[File]` to begin with, but the type sig is `Iterable[Set]` so the compiler doesn't know that.